WalletWasabi/JsonConverters/CompactSignatureJsonConverter.cs

```diff
@@ -0,0 +1,76 @@
+using NBitcoin;
+using Newtonsoft.Json;
+
+namespace WalletWasabi.JsonConverters;
+
+public class CompactSignatureJsonConverter : JsonConverter<CompactSignature>
+{
+	/// <inheritdoc />
+	public override CompactSignature? ReadJson(JsonReader reader, Type objectType, CompactSignature? existingValue, bool hasExistingValue, JsonSerializer serializer)
+	{
+		var value = reader.Value as string;
+
+		if (string.IsNullOrWhiteSpace(value))
+		{
+			return null;
+		}
+
+		byte[] bytes = Convert.FromBase64String(value);
+		int recoveryId = ComputeRecoveryId(bytes[0]);
+		byte[] signature = bytes[1..];
+
+		return new CompactSignature(recoveryId, signature);
+	}
+
+	private static int ComputeRecoveryId(byte b)
+	{
+		int header = b;
+
+		if (header < 27 || header > 34)
```

Comment:
nobody will remember the meaning of these magic numbers. could you try to improve it somehow?